### PR TITLE
Command Line Interface and main function

### PR DESCRIPTION
**Pull request by arthurmloureiro**
This PR introduces a command line interface to conceal sacc data-vectors directly from the terminal. The method accepts inputs from the command line as well as inputs via a yaml configuration file.

The implementation uses the `jsonargparse` library which supports the following command to get help:
```bash
python -m blinding --help
```
or the following command to print an example configuration file:
```bash
python -m blinding --print_config > example_config.yml
```

* Minor bugs in the main `Smokescreen` class were fixed and unit tests were implemented. 
* Fixed some issues with Github actions by updating the `CI.yaml`
* CCL requirements were updated for version `3.0.2` to use the `.to_dict()` method.
* Unit test coverage remains at 95% (unit tests included to cover the main function and other new utility functions too.

### CODE REVIEW

**Comment by tilmantroester** [comment]
The `--print_config` functionality uses some inspection through `jsonargparse`, I assume?

**Comment by arthurmloureiro** [comment]
> The `--print_config` functionality uses some inspection through `jsonargparse`, I assume?

Yes! But it does get a bit confused with the ccl cosmology type.

File: src/blinding/__main__.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        path_to_sacc (str): Path to the sacc file to blind.
+        likelihood_path (str): Path to the firecrown likelihood module file.
+        systematics (dict): Dictionary with fixed values for the firecrown systematics parameters.
+        shifts_dict (dict): Dictionary with fixed values for the firecrown shifts parameters. 
+            Example: {"Omega_c": (0.20, 0.39), "sigma8": (0.70, 0.90)}
+        seed (int, str): Seed for the blinding process. Defaults to 2112.
+        cosmo (Union[CosmologyType, dict]): Cosmology object or dictionary with cosmological parameters you want different than the
+        VanillaLCDM as reference cosmology.
+            Defaults to ccl.CosmologyVanillaLCDM().
+    """
```
Comment (arthurmloureiro) [inline src/blinding/__main__.py:52]: Update this part, it is missing some arguments

**Review by arthurmloureiro — COMMENTED** [review]
Small changes (but I cannot approve my own PR)

**Pull request merged**